media-gfx/gnofract4d: version bump to 4.3

cjmayo opened the pull request:
Package-Manager: Portage-2.3.99, Repoman-2.3.22
Signed-off-by: Chris Mayo <aklhfex@gmail.com>

---

A formal release with fixes that can replace the current snapshot.

Update HOMEPAGE and SRC_URI for the new fract4d organisation
Add missing xdg_pkg_postinst
Remove elog about missing features that gets written even if you don't have missing features.

```diff
--- gnofract4d-4.0.1_p1-r1.ebuild
+++ gnofract4d-4.2.ebuild
@@ -9,15 +9,13 @@
 
 inherit distutils-r1 eutils virtualx xdg
 
-COMMIT=bd600c20921afff7b02fc0a76ab79242ebd0896d
-
 DESCRIPTION="A program for drawing beautiful mathematically-based images known as fractals"
-HOMEPAGE="http://edyoung.github.io/gnofract4d/"
-SRC_URI="https://github.com/edyoung/${PN}/archive/${COMMIT}.tar.gz -> ${P}.tar.gz"
+HOMEPAGE="https://fract4d.github.io/gnofract4d/"
+SRC_URI="https://github.com/fract4d/${PN}/releases/download/v${PV}/${P}.tar.gz"
 
 LICENSE="BSD"
 SLOT="0"
-KEYWORDS="amd64 x86"
+KEYWORDS="~amd64 ~x86"
 IUSE="+doc"
 REQUIRED_USE="test? ( doc )"
 
@@ -43,8 +41,6 @@
 
 distutils_enable_tests pytest
 
-S="${WORKDIR}/${PN}-${COMMIT}"
-
 src_prepare() {
 	distutils-r1_src_prepare
 }
@@ -69,6 +65,6 @@
 }
 
 pkg_postinst() {
-	elog "Optional missing features:"
+	xdg_pkg_postinst
 	optfeature "creating videos" media-video/ffmpeg[vpx,zlib]
 }
```

cjmayo commented:
Replace `PYTHON_MULTI_USEDEP` with  `PYTHON_USEDEP`.

cjmayo commented:
Add testing fract4d_compiler package

```diff
@@ -46,7 +46,7 @@ src_prepare() {
 }
 
 python_test() {
-	virtx pytest fract4d fract4dgui test.py
+	virtx pytest fract4d fract4d_compiler fract4dgui test.py
 }
 
 python_compile_all() {
```

cjmayo commented:
`media-gfx/gnofract4d: add myself as proxy maintainer`

https://github.com/gentoo/gentoo/pull/15987/commits/6374868ca35cf17589f2188946d8a19aa4584410

@gentoo/proxy-maint

cjmayo commented:
Updated to 4.3. New features include no need to build the help, it is now bundled as html.

A bug in the 4.3 release which missed out distributing test.ase.

I've skipped test.py, config.toml is not bundled, but it is really just for upstream checking versions have been updated everywhere.

```diff
--- gnofract4d-4.0.1_p1-r1.ebuild
+++ gnofract4d-4.3.ebuild
@@ -5,70 +5,49 @@
 
 PYTHON_COMPAT=( python3_{6,7,8} )
 DISTUTILS_SINGLE_IMPL=1
-DISTUTILS_USE_SETUPTOOLS=no
 
 inherit distutils-r1 eutils virtualx xdg
 
-COMMIT=bd600c20921afff7b02fc0a76ab79242ebd0896d
-
 DESCRIPTION="A program for drawing beautiful mathematically-based images known as fractals"
-HOMEPAGE="http://edyoung.github.io/gnofract4d/"
-SRC_URI="https://github.com/edyoung/${PN}/archive/${COMMIT}.tar.gz -> ${P}.tar.gz"
+HOMEPAGE="https://fract4d.github.io/gnofract4d/"
+SRC_URI="https://github.com/fract4d/${PN}/releases/download/v${PV}/${P}.tar.gz
+	https://github.com/fract4d/gnofract4d/raw/v4.3/testdata/test.ase -> ${PF}-test.ase"
 
 LICENSE="BSD"
 SLOT="0"
-KEYWORDS="amd64 x86"
-IUSE="+doc"
-REQUIRED_USE="test? ( doc )"
+KEYWORDS="~amd64 ~x86"
 
 COMMON_DEPEND="
 	media-libs/libpng:0=
 	virtual/jpeg:0"
 RDEPEND="${COMMON_DEPEND}
 	$(python_gen_cond_dep '
-		dev-python/pycairo[${PYTHON_MULTI_USEDEP}]
-		dev-python/pygobject:3[${PYTHON_MULTI_USEDEP}]
+		dev-python/pycairo[${PYTHON_USEDEP}]
+		dev-python/pygobject:3[${PYTHON_USEDEP}]
 	')
 	x11-libs/gtk+:3[introspection]"
 BDEPEND="virtual/pkgconfig"
-DEPEND="${COMMON_DEPEND}
-	doc? (
-		app-text/docbook-xsl-stylesheets
-		$(python_gen_cond_dep '
-			dev-python/pygobject[${PYTHON_MULTI_USEDEP}]
-		')
-		dev-libs/libxslt
-		x11-libs/gtk+:3[introspection]
-	)"
+DEPEND="${COMMON_DEPEND}"
 
 distutils_enable_tests pytest
 
-S="${WORKDIR}/${PN}-${COMMIT}"
-
 src_prepare() {
+	cp "${DISTDIR}/${PF}-test.ase" testdata/test.ase || die
 	distutils-r1_src_prepare
+	sed -i -e "s:share/doc/gnofract4d/:share/doc/${PF}/:" setup.py || die
 }
 
 python_test() {
-	virtx pytest fract4d fract4dgui test.py
+	virtx pytest fract4d fract4d_compiler fract4dgui
 }
 
 python_compile_all() {
-	if use doc; then
+	if use test; then
 		ln -s "${BUILD_DIR}"/lib/fract4d/*.so fract4d/ || die
-		"${EPYTHON}" createdocs.py || die
-	fi
-}
-
-python_install_all() {
-	distutils-r1_python_install_all
-	rm -r "${ED}"/usr/share/doc/${PN} || die
-	if ! use doc; then
-		rm -r "${ED}"/usr/share/gnome/help/${PN} || die
 	fi
 }
 
 pkg_postinst() {
-	elog "Optional missing features:"
+	xdg_pkg_postinst
 	optfeature "creating videos" media-video/ffmpeg[vpx,zlib]
 }
```

jsmolic commented:
You can append `[please reassign]` to PR title and the bot will re-assign this to `maintainer-needed` as graphics project is disbanded.\
I suggest you ping someone from proxy maint team to review this, given that you opened this more than 3 months ago :)

jsmolic commented on media-gfx/gnofract4d/gnofract4d-4.3.ebuild:
```diff
@@ -0,0 +1,53 @@
+}
+
+pkg_postinst() {
+	xdg_pkg_postinst
```
This is called automatically in xdg eclass

cjmayo replied:
Yes but pkg_postinst is there because of the optfeature all, which then means the call to xdg_pkg_postinst is needed I think.

jsmolic replied:
Optfeature is handled by the eutils eclass: https://devmanual.gentoo.org/eclass-reference/eutils.eclass/index.html

cjmayo commented:
> You can append `[please reassign]` to PR title and the bot will re-assign this to `maintainer-needed` as graphics project is disbanded.

Thanks for the pointer, it does say that each time - but I never paid attention!
Doubly worth it this time because I added a couple of bug references.

Updated to a snapshot ebuild. It wasn't just the test that was missing from the release tarball, a useful fix as well and since then another significant fix has been merged - the tests should run reliably now.

thesamesam commented on media-gfx/gnofract4d/gnofract4d-4.3_p20200821.ebuild:
```diff
@@ -0,0 +1,55 @@
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
```
Python 3.9 work too?

cjmayo replied:
Yes, 3.9-dev now used in Travis, added.

thesamesam commented on media-gfx/gnofract4d/gnofract4d-4.3_p20200821.ebuild:
```diff
@@ -0,0 +1,55 @@
+PYTHON_COMPAT=( python3_{6,7,8} )
+DISTUTILS_SINGLE_IMPL=1
+
+inherit distutils-r1 eutils virtualx xdg
```
optfeature is now an eclass by itself (see [here](https://archives.gentoo.org/gentoo-dev/message/50778e0b2e8b8b76a6612c5f9b7f1cec), recent change), so should be able to drop eutils.

cjmayo replied:
Changed to optfeature.

thesamesam commented on media-gfx/gnofract4d/gnofract4d-4.3_p20200821.ebuild:
```diff
@@ -0,0 +1,55 @@
+}
+
+python_test() {
+	virtx pytest fract4d fract4d_compiler fract4dgui
```
We may be able to avoid invoking virtx repeatedly by taking [this](https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-python/pyopengl/pyopengl-3.1.5.ebuild?id=f5064a1f8bedef6ed63d5c7131dd0c83a0c3a58a#n53) approach, although a custom python_test is unavoidable it seems.

I also think we [need](https://dev.gentoo.org/~mgorny/python-guide/pytest.html?highlight=pytest%20vv#skipping-tests-based-on-markers) -vv and || die (this is complicated by the virtx though?)

cjmayo replied:
Tried `virtx distutils-r1_src_test` and that works, updated (also moved src_test to have functions in the order of use).

thesamesam commented on media-gfx/gnofract4d/gnofract4d-4.3_p20200821.ebuild:
```diff
@@ -0,0 +1,55 @@
+
+COMMON_DEPEND="
+	media-libs/libpng:0=
+	virtual/jpeg:0"
```
Shouldn't need to specify slot -- there's only one.

cjmayo replied:
Removed the slot.

thesamesam commented on media-gfx/gnofract4d/gnofract4d-4.3_p20200821.ebuild:
```diff
@@ -0,0 +1,55 @@
+	')
+	x11-libs/gtk+:3[introspection]"
+BDEPEND="virtual/pkgconfig"
+DEPEND="${COMMON_DEPEND}"
```
I think we can just drop COMMON_DEPEND and reorder these by setting RDEPEND to be DEPEND plus some bits.

cjmayo replied:
Done.

thesamesam reviewed (commented):
Left some notes. Thanks for your patience on this, it is on my list.

cjmayo commented:
> Left some notes. Thanks for your patience on this, it is on my list.

No problem - some nice fixes have been added in the meantime.

juippis reviewed (commented):
```
    from PIL import Image, ImageChops, ImageStat
E   ModuleNotFoundError: No module named 'PIL'
```
emerging *dev-python/pillow* fixed it. 
But tests seem to hang for me, do they work for you?
It's been here for 30 minutes with nothing happening, no CPU being used etc:
```
fract4dgui/tests/test_browser_model.py::Test::testUpdate PASSED                              [ 90%]
fract4dgui/tests/test_director.py::Test::testDirectorDialog 
```

cjmayo commented:
Thanks for testing.
> ```
>     from PIL import Image, ImageChops, ImageStat
> E   ModuleNotFoundError: No module named 'PIL'
> ```
> 

This comes from test_regress.py which is being collected since the changes to src_test(). It doesn't actually provide any tests so just drop it to avoid the dependency.

> But tests seem to hang for me, do they work for you?
> It's been here for 30 minutes with nothing happening, no CPU being used etc:
> 
> ```
> fract4dgui/tests/test_browser_model.py::Test::testUpdate PASSED                              [ 90%]
> fract4dgui/tests/test_director.py::Test::testDirectorDialog 
> ```

testDirectorDialog is the test for movie making, I suspect this is due to a lack of media-video/ffmpeg[vpx,zlib]. That test dependency is missing, so add that.

Also reorder src_prepare() to put distutils-r1_src_prepare last, don't think it matters here because it is a single implementation but more consistent.

Changes made:

```diff
@@ -20,7 +20,10 @@
 
 DEPEND="
 	media-libs/libpng:0=
-	virtual/jpeg"
+	virtual/jpeg
+	test? (
+		media-video/ffmpeg[vpx,zlib]
+	)"
 RDEPEND="${DEPEND}
 	$(python_gen_cond_dep '
 		dev-python/pycairo[${PYTHON_USEDEP}]
@@ -34,8 +37,11 @@
 S="${WORKDIR}/${PN}-${COMMIT}"
 
 src_prepare() {
+	sed -i -e "s:share/doc/gnofract4d/:share/doc/${PF}/:" setup.py || die
+	# test_regress.py does not provide pytest with any tests and inspecting it requires dev-python/pillow
+	rm test_regress.py || die
+
 	distutils-r1_src_prepare
-	sed -i -e "s:share/doc/gnofract4d/:share/doc/${PF}/:" setup.py || die
 }
 
 python_compile_all() {
```
